RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -654,174 +705,83 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
   } // if (hCalib.isValid()==true)
   if( fillLaserMonitor_  && (hLasermon.isValid() == true) ) {
 
-    std::vector<std::vector<int> > lasmon_adcs(laserMonCBoxList_.size(), std::vector<int>());
-    std::vector<std::vector<int> > lasmon_capids(laserMonCBoxList_.size(), std::vector<int>());
-
-    unsigned max_nsamples = 0;
-    for(QIE10DigiCollection::const_iterator digi = hLasermon->begin(); digi != hLasermon->end(); digi++) {
-
-      QIE10DataFrame df = static_cast<QIE10DataFrame>(*digi);
-
-      HcalCalibDetId calibId( digi->id() );
-
-      // Fill the lasermonitor channels
-      int cboxch  = calibId.cboxChannel( );
-      int iphi    = calibId.iphi();
-      int ieta    = calibId.ieta();
-
-      // only check channels having the requested cboxch
-      // find the index of this channel by matching cBox, iEta, iPhi
-      for( unsigned idx = 0; idx < laserMonCBoxList_.size(); ++idx ) {
-        if( cboxch == laserMonCBoxList_[idx] &&
-          iphi  == laserMonIPhiList_[idx] && 
-          ieta  == laserMonIEtaList_[idx] ) {
+    int icombts = -1;
+    float max_charge = 0;
+    int max_ts = -1;
+    std::vector<float> comb_charge;
+       
+    unsigned nch = laserMonCBoxList_.size();
+    // loop over channels in the order provided
+    for( unsigned ich = 0; ich < nch; ++ich ) {
+      int cboxch = laserMonCBoxList_[ich];
+      int iphi  = laserMonIPhiList_[ich];
+      int ieta  = laserMonIEtaList_[ich];
+
+      // loop over digis, find the digi that matches this channel
+      unsigned nDigi = hLasermon->size();
+      for( unsigned didx = 0; didx < nDigi; ++didx ) {
```

Comment:
When I asked to move to a "range based for loop" [1]  I meant  something like (e.g.):
```
    for(auto & df : hLasermon) {
       HcalCalibDetId calibId( df.id() );
       [...]
    }
```

[1]
http://en.cppreference.com/w/cpp/language/range-for